app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt

```diff
@@ -68,20 +87,35 @@ class CreditCardEditorView(
                 val fields = NewCreditCardFields(
                     billingName = name_on_card_input.text.toString(),
                     plaintextCardNumber = CreditCardNumber.Plaintext(cardNumber),
-                    cardNumberLast4 = cardNumber.substring(cardNumber.length - 4),
+                    cardNumberLast4 = cardNumber.last4Digits(),
                     expiryMonth = (expiry_month_drop_down.selectedItemPosition + 1).toLong(),
                     expiryYear = expiry_year_drop_down.selectedItem.toString().toLong(),
                     cardType = CARD_TYPE_PLACEHOLDER
                 )
                 interactor.onSaveCreditCard(fields)
             }
         }
+    }
 
-        card_number_input.text = state.cardNumber.toEditable()
-        name_on_card_input.text = state.billingName.toEditable()
+    /**
+     * Validates the credit card information entered by user
+     */
+    @VisibleForTesting
+    internal fun validateCreditCardInfo(): Boolean {
```

Comment:
Let's also drop the `Info`

```suggestion
    internal fun validateCreditCard(): Boolean {
```